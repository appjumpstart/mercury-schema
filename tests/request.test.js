const { mercurySchema, ValidationError } = require('../')
const inject = require('light-my-request')

const { userSchema, user } = require('./fixtures')

test('request with valid payload passes validation', async () => {
  try {
    const handler = (req, res) => {
      req.body = JSON.parse(req._lightMyRequest.payload)
      mercurySchema({ request: userSchema })(req, res, () => {
        expect(req.valid).toBe(true)
        res.end()
      })
    }
    await inject(handler, { method: 'POST', url: '/', payload: user })
  } catch (err) {
    fail(err)
  }
})

test('request with missing required property fails validation', async () => {
  try {
    const handler = (req, res) => {
      req.body = JSON.parse(req._lightMyRequest.payload)
      mercurySchema({ request: userSchema })(req, res, () => {
        expect(req.valid).toBe(false)
        expect(req.validationError).toBeInstanceOf(Error)
        expect(req.validationError).toBeInstanceOf(ValidationError)
        const [error] = req.validationError.errors
        expect(error).toMatchSnapshot()
        res.end()
      })
    }
    const payload = { email: user.email }
    await inject(handler, { method: 'POST', url: '/', payload })
  } catch (err) {
    fail(err)
  }
})

test('request with invalid length fails validation', async () => {
  try {
    const handler = (req, res) => {
      req.body = JSON.parse(req._lightMyRequest.payload)
      mercurySchema({ request: userSchema })(req, res, () => {
        expect(req.valid).toBe(false)
        expect(req.validationError).toBeInstanceOf(Error)
        expect(req.validationError).toBeInstanceOf(ValidationError)
        expect(req.validationError.name).toBe('ValidationError')
        const [error] = req.validationError.errors
        expect(error).toMatchSnapshot()
        res.end()
      })
    }
    const payload = { ...user, email: 'c@v.' }
    await inject(handler, { method: 'POST', url: '/', payload })
  } catch (err) {
    fail(err)
  }
})
