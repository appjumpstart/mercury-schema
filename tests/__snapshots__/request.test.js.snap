// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`request with invalid length fails validation 1`] = `
Object {
  "dataPath": ".email",
  "keyword": "minLength",
  "message": "should NOT be shorter than 5 characters",
  "params": Object {
    "limit": 5,
  },
  "schemaPath": "#/properties/email/minLength",
}
`;

exports[`request with missing required property fails validation 1`] = `
Object {
  "dataPath": "",
  "keyword": "required",
  "message": "should have required property 'name'",
  "params": Object {
    "missingProperty": "name",
  },
  "schemaPath": "#/required",
}
`;
